tsdb/engine/tsm1/encoding.go

```diff
@@ -120,7 +120,7 @@ func (v Values) DecodeSameTypeBlock(block []byte) Values {
 // DecodeBlock takes a byte array and will decode into values of the appropriate type
 // based on the block
 func DecodeBlock(block []byte) (Values, error) {
-	if len(block) == 0 {
+	if len(block) < 9 {
```

Comment:
I don't think it makes sense to include this in this PR. Drop this part of it and I'll merge. We can address the encoding stuff elsewhere.